R/score_forecasts.R

```diff
@@ -0,0 +1,97 @@
+#' Score forecasts
+#'
+#' @param forecasts required data.frame with forecasts in the format returned
+#' by load_forecasts
+#' @param truths required data.frame with forecasts in the format returned
+#' by load_truth
+#' @param scores character vector of scores to calculate
+#' @param return_format string: "long" returns long format with a column for
+#' "score_name" and a column for "score_value"; "wide" returns wide format with
+#' a separate column for each score.
+#'
+#' @return data.frame with columns model, forecast_date, location,
+#' target_variable, horizon, target_end_date
+#' If return_format is "long", also contains columns score_name and score_value
+#' where score_name is one of wis, wis_width, wis_penalty_low, wis_penalty_high,
+#' interval_coverage_ where level is 0.95 for a 95% confidence interval, or
+#' quantile_coverage_ where level is 0.975 for a quantile level 0.975
+#' score_value has the numeric value of the score.
+#' If return_format is "wide", each calculated score is in its own column.
+#'
+#' @export
+score_forecasts <- function(
+  forecasts,
+  truth,
+  scores = vector(),
+  return_format = "wide"
+) {
+  
+  # forecasts data.frame format
+  # columns: model, forecast_date, location, horizon, temporal_resolution,
+  #          target_variable, target_end_date, type, quantile, value
+  forecasts_colnames <- c(
+    "model", "forecast_date", "location", "horizon", "temporal_resolution",
+    "target_variable", "target_end_date", "type", "quantile", "value"
+  )
+  
+  # validate forecasts
+  if (missing(forecasts) || is.null(forecasts)) {
+    stop("Forecast dataframe missing", call. = TRUE)
+  } else if (!setequal(colnames(forecasts), forecasts_colnames)) {
```

Comment:
Note that load_forecasts has been updated and now there is a larger set of geospatial columns available. Maybe we could just require the column names to have the necessary columns as opposed to checking that they are all there?

Reply:
Note that this applies to truth data format as well.

Reply:
Yep, I can make the conditional into a subset operation.